src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,16 +209,17 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                     //
                     // Avoid instance state modification
-                    Dictionary<string, ConfigurationSetting> applicationSettings = null;
+                    Dictionary<string, ConfigurationSetting> serverData = null;
                     Dictionary<KeyValueWatcher, KeyValueChange> keyValueChanges = null;
                     List<KeyValueChange> changedKeyValuesCollection = null;
+                    Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(_mappedData);

```

Comment:
> Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(_mappedData); [](http://example.com/codeflow?start=19&length=113)

It would be better to keep the same initialization pattern here as serverData. Since both dictionaries need to be updated at the same time, keeping their code style in sync would help with readability.
We can create a new dictionary for mappedData when its needed (we do the same for serverData later).